project_template/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.

```

Comment:
Is there a way to share parts of this file across different projects? Something like `include` ?

Reply:
yes, however it's frequently not a good idea. The stuff on line 15 will add cmake/ to the module path and allow including things from there.

You can also do what extra-cmake-modules does and maintain that stuff as a seperate project.

The issue with having a parent cmakelists file set stuff like visibility and whatnot is that it means the child isn't an independent project anymore, which can be OK, but is something to balance.

Reply:
IMHO, for Azure SDK we would like to have it because we may have about 100 of projects in Azure SDK for C. So, to be consistent, we would like to share code/scripts/configuration as much as possible between Azure SDK projects.

Reply:
@bsiegel could you share your experience on this subjects?

Reply:
might we want to try and limit the shared configurations/code/scripts to things only absolutely necessary across all repos? say we have the c and cpp standards set in the top level since it is a requirement across all azure c repos. but any other config attempting to be put in the top level should have strong justification since it will affect all repos.

Reply:
IMHO what we should have is something like extra-cmake-modules in the "core" repo.

So some core project (which we'll probably end up with anyway) will provide cmake macros/functions and will be a build-dependency of all/most other azure-c projects.

Projects could say something like
```cmake
if(NOT DEFINED AZ_CORE_MODULE_PATH)
    find_package(az_core REQUIRED CONFIG)
endif()
list(APPEND CMAKE_MODULE_PATH ${AZ_CORE_MODULE_PATH})
```

then do as they need.

@ras0219-msft what do you think about this? Is this a good way to go about it?